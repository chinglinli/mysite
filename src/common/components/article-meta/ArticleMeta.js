import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'
import dateformat from 'dateformat'

import styles from './ArticleMeta.scss'

export default class ArticleMeta extends React.PureComponent {
  static PropTypes = {
    categories: ImmutablePropTypes.list,
    date: PropTypes.any.isRequired,
    modifiedDate: PropTypes.any
  }

  render () {
    const {categories, date, modifiedDate} = this.props

    return (
      <div className={styles.root}>
        <span className={styles.field}>
          分類：{ categories.map(category => (
          <Link key={category.get('slug')}
                className={styles.category}
                to={`/articles/category/${category.get('slug')}/`}>{category.get('name')}</Link>
        )) }
        </span>
        <span className={styles.field}>作者：兩大類</span>
        <span className={styles.field}>發表時間：{dateformat(date, 'yyyy/mm/dd')}</span>
        {modifiedDate && <span className={styles.field}>最後更新：{dateformat(modifiedDate, 'yyyy/mm/dd')}</span>}
      </div>
    )
  }
}
