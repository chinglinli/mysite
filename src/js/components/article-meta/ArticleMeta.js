import * as React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Link } from 'react-router';

import styles from './ArticleMeta.scss';


export default class ArticleMeta extends React.Component {
    static PropTypes = {
        categories: ImmutablePropTypes.list,
        date: React.PropTypes.any.isRequired,
        modifiedDate: React.PropTypes.any
    };

    render() {
        const { categories, date, modifiedDate } = this.props;

        return (
            <div className={styles.root}>
                <span className={styles.field}>
                    分類：{ categories.map(category => (
                        <Link key={category.get('slug')}
                              className={styles.category}
                              to={`/articles/category/${category.get('slug')}/`}>{category.get('name')}</Link>
                    )) }
                </span>
                <span className={styles.field}>
                    作者：兩大類
                </span>
                <span className={styles.field}>
                    發表時間：{ date.format('YYYY/MM/DD') }
                </span>
                {modifiedDate &&
                    <span className={styles.field}>
                        最後更新：{ modifiedDate.format('YYYY/MM/DD') }
                    </span>
                }
            </div>
        );
    }
}
