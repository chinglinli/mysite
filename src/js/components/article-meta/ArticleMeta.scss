.root {
    margin-top: 0.5em;
}

.field {
    font-size: .9em;
    &:not(:first-child) {
        margin-left: 1.5em;
    }
}

.category:not(:first-child):before {
    content: ' / ';
}