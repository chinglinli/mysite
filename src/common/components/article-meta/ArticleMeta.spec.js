import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import ArticleMeta from './ArticleMeta'


test('It should render ArticleMeta component without modifiedDate correctly', () => {
  const props = Immutable.fromJS({
    categories: [
      {slug: 'slug 1', name: 'category 1'}
    ],
    date: new Date(2017, 4, 15),
    modifiedDate: null
  })

  const component = renderer.create(
    <ArticleMeta
      categories={props.get('categories')}
      date={props.get('date')}
      modifiedDate={props.get('modifiedDate')}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})

test('It should render ArticleMeta component with multiple categories correctly', () => {
  const props = Immutable.fromJS({
    categories: [
      {slug: 'slug 1', name: 'category 1'},
      {slug: 'slug 2', name: 'category 2'}
    ],
    date: new Date(2017, 4, 15),
    modifiedDate: null
  })

  const component = renderer.create(
    <ArticleMeta
      categories={props.get('categories')}
      date={props.get('date')}
      modifiedDate={props.get('modifiedDate')}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})


test('It should render ArticleMeta component correctly', () => {
  const props = Immutable.fromJS({
    categories: [
      {slug: 'slug 1', name: 'category 1'},
      {slug: 'slug 2', name: 'category 2'}
    ],
    date: new Date(2017, 4, 15),
    modifiedDate: new Date(2017, 4, 16)
  })

  const component = renderer.create(
    <ArticleMeta
      categories={props.get('categories')}
      date={props.get('date')}
      modifiedDate={props.get('modifiedDate')}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
