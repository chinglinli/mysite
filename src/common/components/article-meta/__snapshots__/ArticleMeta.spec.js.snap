// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render ArticleMeta component correctly 1`] = `
<div
  className="root"
>
  <span
    className="field"
  >
    分類：
    <a
      className="category"
      onClick={[Function]}
      style={Object {}}
    >
      category 1
    </a>
    <a
      className="category"
      onClick={[Function]}
      style={Object {}}
    >
      category 2
    </a>
  </span>
  <span
    className="field"
  >
    作者：兩大類
  </span>
  <span
    className="field"
  >
    發表時間：
    2017/05/15
  </span>
  <span
    className="field"
  >
    最後更新：
    2017/05/16
  </span>
</div>
`;

exports[`It should render ArticleMeta component with multiple categories correctly 1`] = `
<div
  className="root"
>
  <span
    className="field"
  >
    分類：
    <a
      className="category"
      onClick={[Function]}
      style={Object {}}
    >
      category 1
    </a>
    <a
      className="category"
      onClick={[Function]}
      style={Object {}}
    >
      category 2
    </a>
  </span>
  <span
    className="field"
  >
    作者：兩大類
  </span>
  <span
    className="field"
  >
    發表時間：
    2017/05/15
  </span>
</div>
`;

exports[`It should render ArticleMeta component without modifiedDate correctly 1`] = `
<div
  className="root"
>
  <span
    className="field"
  >
    分類：
    <a
      className="category"
      onClick={[Function]}
      style={Object {}}
    >
      category 1
    </a>
  </span>
  <span
    className="field"
  >
    作者：兩大類
  </span>
  <span
    className="field"
  >
    發表時間：
    2017/05/15
  </span>
</div>
`;
